src/common/config.cc

```diff
@@ -786,45 +811,30 @@ int md_config_t::_get_val(const char *key, char **buf, int len) const
   // In key names, leading and trailing whitespace are not significant.
   string k(ConfFile::normalize_key_name(key));
 
-  for (int i = 0; i < NUM_CONFIG_OPTIONS; ++i) {
-    const config_option *opt = &config_optionsp[i];
-    if (strcmp(opt->name, k.c_str()))
-      continue;
+  auto opt_it = config_options.find(k);
+  if (config_options.end() == opt_it) {
+      return config_value_t(invalid_config_value_t());
+  }
+  get_value_generic_visitor gvv(this);
+  return boost::apply_visitor(gvv, opt_it->second.md_member_ptr);
+}
 
+int md_config_t::_get_val(const char *key, char **buf, int len) const
+{
+  assert(lock.is_locked());
+
+  if (!key)
+    return -EINVAL;
```

Comment:
returning -EINVAL here is wrong since it looks like as a valid integer valued option